src/hotspot/cpu/s390/templateTable_s390.cpp

```diff
@@ -76,10 +77,13 @@
   __ bind(lbl);                                                                \
   { unsigned int b_off = __ offset();                                          \
     uintptr_t   b_addr = (uintptr_t)__ pc();                                   \
-    __ z_larl(Z_R0, (int64_t)0);     /* Check current address alignment. */    \
-    __ z_slgr(Z_R0, br_tab);         /* Current Address must be equal    */    \
-    __ z_slgr(Z_R0, flags);          /* to calculated branch target.     */    \
-    __ z_brc(Assembler::bcondLogZero, 3); /* skip trap if ok. */               \
+    __ z_larl(br_tab_temp, (int64_t)0);  /* Check current address alignment. */\
+    __ z_slgr(br_tab_temp, br_tab);      /* Current Address must be equal    */\
+    if (tos_state != Z_R0_scratch) {                                           \
```

Comment:
Where can tos_state be R0?

Reply:
This was an intermediate state. In my review (not complete yet) I request this special case to be removed.